store/mongo/datastore_mongo.go

```diff
@@ -1582,9 +1602,18 @@ func (db *DataStoreMongo) GetDevicesListForDeployment(ctx context.Context,
 	}
 
 	options := mopts.Find()
-	sortFieldQuery := bson.D{
-		{Key: StorageKeyDeviceDeploymentStatus, Value: 1},
-		{Key: StorageKeyDeviceDeploymentDeviceId, Value: 1},
+	var sortFieldQuery bson.D
+	if q.SortByArtifactSize {
+		sortFieldQuery = bson.D{
+			{Key: StorageKeyDeviceDeploymentArtifactSize, Value: -1},
+			{Key: StorageKeyDeviceDeploymentStatus, Value: 1},
+			{Key: StorageKeyDeviceDeploymentDeviceId, Value: 1},
```

Comment:
Do we really get value sorting by artifact size, status and device ID?
The index will be quite large (three fields), and at the end of the day what you want to do is sorting by artifact size and keep everything else stable. Maybe artifact size and deviceid is enough?